### ISSUE TITLE
ERROR: Failed to run fusesoc:utils:blinky : '['make', 'pgm']' exited with an error: 2

### ISSUE THREAD

**Issue by rafaelnp** [olofk/fusesoc#526]
I am using Fusesoc 1.12.0  with  Vivado 2020.2 and 2021.1 und  I am having a problem on Windows 10  to programme the FPGA. I used the blinky example  from  [blinky](https://github.com/fusesoc/blinky) repositoty as minimal/simplesta reproducable case, to generate the bitstream file and programme the Digilent Basys3 board.

Additional Infos:
- Python 3.10
- Windows 10 20H2 (Build 19042.1288)
- GNU Make 4.3 (Installed via Chocolatey)

I ran:
```
fusesoc run --target=basys3 fusesoc:utils:blinky
```

then I get the following error:
```
ERROR: Failed to run fusesoc:utils:blinky : '['make', 'pgm']' exited with an error: 2
```

When trying to discover what would be the root cause, I  observed that the only step that failed is the last one, programming the FPGA. Running:

```
fusesoc run --target=basys3 --run fusesoc:utils:blinky
```

Results in the same error as before. Additionally I checked if Make is present in the PATH. On (Arch) Linux I do not experience this problem.

Maybe I am mssing some configuration and I am not aware of it.  Any thoughts or help would be appreciated. Thanks in advance.

**Comment by olofk** [comment]
@rafaelnp Did you ever figure out what the root cause was? The only thing I can think of is if calling from Python means different paths get used. What happens if you put this
```
import subprocess
subprocess.run(['make', 'pgm'])
```
in a python script in your work root and run it instead of running `make pgm` directly?

**Comment by rafaelnp** [comment]
@olofk I could not figure out, so far, what is the root cause. I checked the PATH variable, I tried to debug and no culprit found. I ran the script you suggest, and I saw the same error.  This line `zsh: exec format error: vivado` intrigues me...
Is there code in fusesoc depending on an environment variable somehow?

**Comment by olofk** [comment]
@rafaelnp If the same problem happens with the python script I wrote in the comment, then I think something is up with your environment which is hard to understand from the outside. My first thought would be to look where your `make` command comes from. It feels like that could potentially be part of some other ecosystem on your computer and that is why things gets mixed up

**Comment by rafaelnp** [comment]
@olofk thanks for your feedback. I will try to setup a second windows 10 computer, and try to reproduce the problem on it.

**Comment by rafaelnp** [comment]
@olofk @imphil sorry for the late reply, it took me a while to get new windows 10 computer to reproduce the problem. In fact I got 2 computers, and could not reproduce the error. Both got new clean Windows 10 installed, and then I installed Vivvado, Python, Fusecos, and so on.. It is  quite annoyng not being able to find the root cause. So far it goes in the direction of a "corrupt" environment in the computer in questions. I took a further look, and could not narrow it down. As it is not a Fusesoc problem, I an gonna close this issue.
